File: ChainOfResponsibility/JavaExample/ChainOfResponsibility/src/SilverHandler.java
public class SilverHandler extends BaseDiscountHandler{
    @Override
    public void getDiscount(String clientType) {
        if ("silver".equalsIgnoreCase(clientType)) {
            System.out.println("Silver client gets 10% discount.");
        } else {
            next(clientType);
        }
    }
}

File: ChainOfResponsibility/ChainOfResponsibilityShop/ChainOfResponsibilityShop/SilverHandler.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ChainOfResponsibilityShop
{
    public class SilverHandler : BaseDiscountHandler 
    {
        public override void GetDiscount(string clientType)
        {
            if (clientType == "silver" )
            {
                Console.WriteLine("Silver client gets 10% discount.");
            }
            else 
            {
                Next(clientType);
            }
        }
    }
}
